### PR TITLE
Switch the embedded JDK to version 17.

### PR DESCRIPTION
**Pull request by benjaminp**
This allows removing the "cached" allmodules and minimized jdk archives, since `jlink` is deterministic in newer JDK versions.

Fixes https://github.com/bazelbuild/bazel/issues/14785.

### CODE REVIEW

File: src/test/shell/integration/bazel_command_log_test.sh [hunk 1]
```diff
@@ -45,6 +45,7 @@ function strip_lines_from_bazel_cc() {
     -e '/^WARNING: Waiting for server process to terminate (waited 5 seconds, waiting at most 60)$/d' \
     -e '/^WARNING: The startup option --host_javabase is deprecated; prefer --server_javabase.$/d' \
     -e '/^WARNING: The home directory is not defined, no home_rc will be looked for.$/d' \
+    -e '/Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release/d' \
```
Comment (cushon) [inline src/test/shell/integration/bazel_command_log_test.sh:48]: Should https://github.com/bazelbuild/bazel/issues/13097 be addressed first, to avoid log spam? I'm fine with just removing `-Xverify:none` at this point

File: src/minimize_jdk.sh [hunk 2]
```diff
@@ -79,7 +79,8 @@ else
   else
     cp lib/libdt_socket.so lib/libjdwp.so reduced/lib
   fi
-  GZIP=-9 tar -zcf ../reduced.tgz reduced
+  find reduced -exec touch -ht 198001010000 {} +
+  zip -q -X -r ../reduced.zip reduced/
```
Comment (cushon) [inline src/minimize_jdk.sh:83]: Just curious, why the change from tar to zip?

**Review by cushon — CHANGES_REQUESTED** [review]
Thanks! Upgrading to JDK 17 here SGTM overall.

File: src/test/shell/integration/bazel_command_log_test.sh (hunk 1 above)
Comment (benjaminp) [reply src/test/shell/integration/bazel_command_log_test.sh:48]: Okay—I had the impression that the logspam was preferred over the performance hit from bytecode verification.

I should just delete this line? https://github.com/bazelbuild/bazel/blob/a67c823938dcc426952d25b3a68b825cbe5f6586/src/main/cpp/blaze.cc#L370

Comment (meisterT) [reply src/test/shell/integration/bazel_command_log_test.sh:48]: @benjaminp When exactly do we print the line?

**Review by meisterT — APPROVED** [review]
Awesome that `jlink` is deterministic now. That is so much better.

**Comment by benjaminp** [comment]
I suspect the `//third_party/ijar/test:ijar_test` failure is a preëxisting flake. See https://github.com/bazelbuild/bazel/pull/16788.

**Comment by meteorcloudy** [comment]
This looks like a nice improvement, should we cherry pick to Bazel 6.0? Not sure if there are more dependency changes if we do that.

**Comment by meisterT** [comment]
Just discussed with @meteorcloudy in person and decided that we should not cherry pick this to not delay the release.

**Comment by meteorcloudy** [comment]
Looks like this broke our cross compile for Bazel arm64 binary on Apple Silicon
```
bazel build --show_progress_rate_limit=5 --curses=yes --color=yes --terminal_columns=143 --show_timestamps --verbose_failures --jobs=12 --announce_rc --experimental_repository_cache_hardlinks --disk_cache= --sandbox_writable_path=/var/tmp/_bazel_buildkite/cache/repos/v1 --test_env=REPOSITORY_CACHE=/var/tmp/_bazel_buildkite/cache/repos/v1 -c opt --cpu=darwin_arm64 --test_env=HOME --test_env=BAZELISK_USER_AGENT -- //src:bazel //src:bazel_nojdk
(16:31:26) INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=0
(16:31:26) INFO: Reading rc options for 'build' from /Users/buildkite/builds/bk-trusted-macpro-1/bazel-trusted/publish-bazel-binaries/.bazelrc:
  Inherited 'common' options: --enable_platform_specific_config
(16:31:26) INFO: Reading rc options for 'build' from /Users/buildkite/builds/bk-trusted-macpro-1/bazel-trusted/publish-bazel-binaries/.bazelrc:
  'build' options: --java_language_version=11 --tool_java_language_version=11
(16:31:26) INFO: Found applicable config definition build:macos in file /Users/buildkite/builds/bk-trusted-macpro-1/bazel-trusted/publish-bazel-binaries/.bazelrc: --macos_minimum_os=10.10
(16:31:26) INFO: Current date is 2022-11-18
(16:31:56) INFO: Analyzed 2 targets (379 packages loaded, 10431 targets configured).
(16:31:56) INFO: Found 2 targets...
(16:32:04) ERROR: /Users/buildkite/builds/bk-trusted-macpro-1/bazel-trusted/publish-bazel-binaries/src/BUILD:175:8: Executing genrule //src:embedded_jdk_minimal failed: (Exit 1): bash failed: error executing command
  (cd /private/var/tmp/_bazel_buildkite/9b9d23816be6e419cbff7fb0e6340516/sandbox/darwin-sandbox/103/execroot/io_bazel && \
  exec env - \
    PATH=/Users/buildkite/Library/Caches/bazelisk/downloads/bazelbuild/bazel-5.3.2-darwin-x86_64/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin \
  /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh; src/minimize_jdk.sh external/openjdk_macos_aarch64_vanilla/file/zulu-macos-aarch64-vanilla.tar.gz src/jdeps_modules.golden bazel-out/darwin_arm64-opt/bin/src/minimal_jdk.zip')
# Configuration: d5bf03822cc4f79e3333829bd4de54a9077ad0ac9deac3f8d62407f76e1aa6c5
# Execution platform: //:default_host_platform
 
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
src/minimize_jdk.sh: line 71: ./bin/jlink: Bad CPU type in executable
```

**Comment by meteorcloudy** [comment]
Fixed by changing the `macos_arm64` platform to actually use the Apple Silicon machines. ;)
